### ISSUE TITLE
Error with LDAP auth - "Error authenticating via omniauth: undefined method `[]' for nil:NilClass""

### ISSUE THREAD

**Issue by dsebdlf** [bigbluebutton/greenlight#213]
Hi!

Trying to login using our ldap server, i'm geting this error:

> "Error authenticating via omniauth: undefined method `[]' for nil:NilClass"

Greenlight is running in a docker container; using ldap-utils from inside the container I can connect correctly to the ldap server.

Activating debug, i get this error in the production.log:

> I, [2018-04-20T17:38:53.684179 #1]  INFO -- : [1dc0a76c-84d5-4b47-bb56-497d845218c7]   Parameters: {"username"=>"someuser", "password"=>"[FILTERED]", "provider"=>"ldap"}
> D, [2018-04-20T17:38:53.686684 #1] DEBUG -- : [1dc0a76c-84d5-4b47-bb56-497d845218c7]   User Load (0.5ms)  SELECT  "users".* FROM "users" WHERE "users"."uid" = ? AND "users"."provider" = ? LIMIT ?  [["uid", "mail=someuser@test.com,vd=test.com,dc=test,dc=com"], ["provider", "ldap"], ["LIMIT", 1]]
> D, [2018-04-20T17:38:53.712051 #1] DEBUG -- : [1dc0a76c-84d5-4b47-bb56-497d845218c7]    (0.1ms)  begin transaction
> D, [2018-04-20T17:38:53.713608 #1] DEBUG -- : [1dc0a76c-84d5-4b47-bb56-497d845218c7]    (0.1ms)  rollback transaction
> E, [2018-04-20T17:38:53.715177 #1] ERROR -- : [1dc0a76c-84d5-4b47-bb56-497d845218c7] Error authenticating via omniauth: undefined method `[]' for nil:NilClass
> 

In the env file i got the following in the ldap section:

> LDAP_SERVER=10.x.x.x
> LDAP_PORT=389
> LDAP_METHOD=plain
> LDAP_UID=username
> LDAP_BASE=dc=test,dc=com
> LDAP_BIND_DN=cn=Manager,dc=test,dc=com
> LDAP_PASSWORD=****

In the browser, the login form redirects to greenlight home page, no error nor messages.

Any hint?

thanks!

**Comment by joshua-arts** [comment]
Hey @dsebdlf,

The error is occurring when Greenlight is attempting to [create the user from omniauth](https://github.com/bigbluebutton/greenlight/blob/a45f4d4fd02e53de5e04180ffb9fdf42bcf2e5c2/app/models/user.rb#L24).

On the login screen, are you inputing `"mail=someuser@test.com,vd=test.com,dc=test,dc=com"` for the username? You should only need to input the users email.

If the above doesn't fix your problem, what LDAP software are you using?

**Comment by dsebdlf** [comment]
Hi @joshua-arts! 

I'm using OpenLdap 2.3.34, and in the login screen I use the mail or username (those are the ldap fields that we use to login in other services), I've tried both of them, with the correspondent change in the env file. I don't know why the log shows that information.
The ldap server is a read only replica of our master, so greenlight can't create users, and of course this is not desirable.

I've tried googe auth, it works fine, but we need to restrict the use of BBB only to our users. We're disabling guest access as well

thanks!
